Pentaho PDI use Native JDBC TO ClickHouse  net/jpountz/lz4/LZ4Factory

Error connecting to database [ClickHouse] :org.pentaho.di.core.exception.KettleDatabaseException: 
Error occurred while trying to connect to the database

Error connecting to database: (using class com.github.housepower.jdbc.ClickHouseDriver)
net/jpountz/lz4/LZ4Factory


org.pentaho.di.core.exception.KettleDatabaseException: 
Error occurred while trying to connect to the database

Error connecting to database: (using class com.github.housepower.jdbc.ClickHouseDriver)
net/jpountz/lz4/LZ4Factory


	at org.pentaho.di.core.database.Database.normalConnect(Database.java:472)
	at org.pentaho.di.core.database.Database.connect(Database.java:370)
	at org.pentaho.di.core.database.Database.connect(Database.java:341)
	at org.pentaho.di.core.database.Database.connect(Database.java:331)
	at org.pentaho.di.core.database.DatabaseFactory.getConnectionTestReport(DatabaseFactory.java:80)
	at org.pentaho.di.core.database.DatabaseMeta.testConnection(DatabaseMeta.java:2786)
	at org.pentaho.ui.database.event.DataHandler.testDatabaseConnection(DataHandler.java:619)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.pentaho.ui.xul.impl.AbstractXulDomContainer.invoke(AbstractXulDomContainer.java:313)
	at org.pentaho.ui.xul.impl.AbstractXulComponent.invoke(AbstractXulComponent.java:157)
	at org.pentaho.ui.xul.impl.AbstractXulComponent.invoke(AbstractXulComponent.java:141)
	at org.pentaho.ui.xul.swt.tags.SwtButton.access$500(SwtButton.java:43)
	at org.pentaho.ui.xul.swt.tags.SwtButton$4.widgetSelected(SwtButton.java:137)
	at org.eclipse.swt.widgets.TypedListener.handleEvent(Unknown Source)
	at org.eclipse.swt.widgets.EventTable.sendEvent(Unknown Source)
	at org.eclipse.swt.widgets.Display.sendEvent(Unknown Source)
	at org.eclipse.swt.widgets.Widget.sendEvent(Unknown Source)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Unknown Source)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Unknown Source)
	at org.eclipse.jface.window.Window.runEventLoop(Window.java:820)
	at org.eclipse.jface.window.Window.open(Window.java:796)
	at org.pentaho.di.ui.xul.KettleDialog.show(KettleDialog.java:80)
	at org.pentaho.di.ui.xul.KettleDialog.show(KettleDialog.java:47)
	at org.pentaho.di.ui.core.database.dialog.XulDatabaseDialog.open(XulDatabaseDialog.java:118)
	at org.pentaho.di.ui.core.database.dialog.DatabaseDialog.open(DatabaseDialog.java:60)
	at org.pentaho.di.ui.spoon.delegates.SpoonDBDelegate.editConnection(SpoonDBDelegate.java:95)
	at org.pentaho.di.ui.spoon.Spoon.doubleClickedInTree(Spoon.java:3202)
	at org.pentaho.di.ui.spoon.Spoon.doubleClickedInTree(Spoon.java:3137)
	at org.pentaho.di.ui.spoon.Spoon.access$2200(Spoon.java:370)
	at org.pentaho.di.ui.spoon.Spoon$29.widgetDefaultSelected(Spoon.java:6286)
	at org.eclipse.swt.widgets.TypedListener.handleEvent(Unknown Source)
	at org.eclipse.swt.widgets.EventTable.sendEvent(Unknown Source)
	at org.eclipse.swt.widgets.Display.sendEvent(Unknown Source)
	at org.eclipse.swt.widgets.Widget.sendEvent(Unknown Source)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Unknown Source)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Unknown Source)
	at org.pentaho.di.ui.spoon.Spoon.readAndDispatch(Spoon.java:1376)
	at org.pentaho.di.ui.spoon.Spoon.waitForDispose(Spoon.java:8161)
	at org.pentaho.di.ui.spoon.Spoon.start(Spoon.java:9523)
	at org.pentaho.di.ui.spoon.Spoon.main(Spoon.java:702)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.pentaho.commons.launcher.Launcher.main(Launcher.java:92)
Caused by: org.pentaho.di.core.exception.KettleDatabaseException: 
Error connecting to database: (using class com.github.housepower.jdbc.ClickHouseDriver)
net/jpountz/lz4/LZ4Factory

	at org.pentaho.di.core.database.Database.connectUsingClass(Database.java:585)
	at org.pentaho.di.core.database.Database.normalConnect(Database.java:456)
	... 47 more
Caused by: java.lang.NoClassDefFoundError: net/jpountz/lz4/LZ4Factory
	at com.github.housepower.jdbc.buffer.CompressedBuffedWriter.<init>(CompressedBuffedWriter.java:16)
	at com.github.housepower.jdbc.serializer.BinarySerializer.<init>(BinarySerializer.java:21)
	at com.github.housepower.jdbc.connect.PhysicalConnection.openPhysicalConnection(PhysicalConnection.java:131)
	at com.github.housepower.jdbc.ClickHouseConnection.createPhysicalInfo(ClickHouseConnection.java:173)
	at com.github.housepower.jdbc.ClickHouseConnection.createClickHouseConnection(ClickHouseConnection.java:169)
	at com.github.housepower.jdbc.NonRegisterDriver.connect(NonRegisterDriver.java:23)
	at java.sql.DriverManager.getConnection(DriverManager.java:664)
	at java.sql.DriverManager.getConnection(DriverManager.java:208)
	at org.pentaho.di.core.database.Database.connectUsingClass(Database.java:567)
	... 48 more
Caused by: java.lang.ClassNotFoundException: net.jpountz.lz4.LZ4Factory
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 57 more

Custom URL     :jdbc:clickhouse://IP:9000/default
Custom Driver Class:com.github.housepower.jdbc.ClickHouseDriver

--- a/pom.xml
+++ b/pom.xml
@@ -320,22 +320,15 @@
                         </goals>
 
                         <configuration>
+                            <transformers>
+                                <transformer implementation="org.apache.maven.plugins.shade.resource.ComponentsXmlResourceTransformer"/>
+                            </transformers>
+
                             <shadedArtifactAttached>true</shadedArtifactAttached>
                             <relocations>
                                 <relocation>
-                                    <pattern>com.</pattern>
-                                    <shadedPattern>com.github.housepower.repackaged.com</shadedPattern>
-                                    <excludes>
-                                        <exclude>com.github.housepower.jdbc.*</exclude>
-                                    </excludes>
-                                </relocation>
-
-                                <relocation>
                                     <pattern>net.</pattern>
                                     <shadedPattern>com.github.housepower.repackaged.net</shadedPattern>
-                                    <excludes>
-                                        <exclude>net.jpountz.lz4.*</exclude>
-                                    </excludes>
                                 </relocation>
 
                                 <relocation>

--- a/pom.xml
+++ b/pom.xml
@@ -320,15 +320,22 @@
                         </goals>
 
                         <configuration>
+                            <shadedArtifactAttached>true</shadedArtifactAttached>
                             <relocations>
                                 <relocation>
                                     <pattern>com.</pattern>
                                     <shadedPattern>com.github.housepower.repackaged.com</shadedPattern>
+                                    <excludes>
+                                        <exclude>com.github.housepower.jdbc.*</exclude>
+                                    </excludes>
                                 </relocation>
 
                                 <relocation>
                                     <pattern>net.</pattern>
                                     <shadedPattern>com.github.housepower.repackaged.net</shadedPattern>
+                                    <excludes>
+                                        <exclude>net.jpountz.lz4.*</exclude>
+                                    </excludes>
                                 </relocation>
 
                                 <relocation>
@@ -336,7 +343,6 @@
                                     <shadedPattern>com.github.housepower.repackaged.org</shadedPattern>
                                 </relocation>
                             </relocations>
-                            <shadedArtifactAttached>true</shadedArtifactAttached>
                         </configuration>
                     </execution>
                 </executions>

